syntax = "proto3";

service Update {
    rpc Get(GetRequest) returns (GetResponse) {}
    rpc Put(PutRequest) returns (PutResponse) {}
    rpc CommitQuery(CommitQueryMessage) returns (CommitQueryResponse) {}
    rpc Commmit(CommitMessage) returns (CommitResponse) {}
}

message GetRequest {
    int32 id = 1;
    string key = 2;
}

message GetResponse {
    int32 value = 1;
}


message PutRequest {
    int32 id = 1;
    string key = 2;
    int32 value = 3;
}

message PutResponse {
    string empty = 1;
}

message CommitQueryMessage {
    int32 id = 1;
    int32 commit_id = 2;
}

message CommitQueryResponse {
    string ack = 1;
}

message CommitMessage {
    int32 id = 1;
}

message CommitResponse {
    string ack = 1;
} 